nav2_map_server/src/map_saver.cpp

```diff
@@ -1,110 +1,203 @@
-/*
- * map_saver
- * Copyright (c) 2008, Willow Garage, Inc.
- * All rights reserved.
- *
- * Redistribution and use in source and binary forms, with or without
- * modification, are permitted provided that the following conditions are met:
- *
- *     * Redistributions of source code must retain the above copyright
- *       notice, this list of conditions and the following disclaimer.
- *     * Redistributions in binary form must reproduce the above copyright
- *       notice, this list of conditions and the following disclaimer in the
- *       documentation and/or other materials provided with the distribution.
- *     * Neither the name of the <ORGANIZATION> nor the names of its
- *       contributors may be used to endorse or promote products derived from
- *       this software without specific prior written permission.
- *
- * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
- * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
- * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
- * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
- * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
- * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
- * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
- * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
- * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
- * POSSIBILITY OF SUCH DAMAGE.
- */
+// Copyright 2019 Rover Robotics
+// Copyright 2008, Willow Garage, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
I'm happy to do either, but the license in package.xml is Apache. Should I add a second license to package.xml?

Reply:
https://github.com/ros-planning/navigation/blob/melodic-devel/map_server/package.xml#L17

It was definitely BSD, as was most of navigation1, I think that got messed up in transition, we should revert it back to BSD for Willow and you're more than welcome to have it be Apache for your changes

Reply:
Reverted it to BSD. I think it was probably a conscientious choice to move everything to one license, but I'll let @mkhansen-intel or someone else figure that out and make sure all the licenses are in order.

Reply:
Makes sense, but we also can't overwrite history, feel free to not resolve in this PR if we file a separate ticket, it really shouldnt block you if you didnt put it there

Reply:
It should stay BSD. If it's Apache in the package.xml, then that's a bug. We can't legally change the licenses of the BSD files. We have been using Apache for any "new" code files though. So we  have a mixture in the repo.

Also, we haven't been adding any additional copyrights for those files that are already copyrighted. I'm not sure how legally that can be done either.

Reply:
kk. Added BSD to package.xml. I'm adding a copyright to files that have substantially changed, which I think is correct - I don't want to imply that my contributions are from Willow Garage, Inc, which wrote a minority of the code that is now in map_saver.cpp.

In any case, ideally, we wouldn't waste 30 lines at the top of a file to a license (which also makes an easy place for someone to sneak in a nefarious extra clause), but have something in the repo that says "unless otherwise credited, all source files are licensed under XXX"